### PR TITLE
✨adding `dynamic-scheduler` service (⚠️ devops)

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

- The `dynamic-scheduler` will gradually be populated with the internals of the dynamic scheduler from director-v2. Code will gradually move into this service.
- New functionality related to dynamic services will end un inside this service.
- This will put the scalability of the dynamic backend in reach.

# ⚠️ devops

New service needs to be deployed with the stack

- [ ] deployment constraints (service runs with 2 copies on all deployments)

Captured env vars
```bash
LOG_FORMAT_LOCAL_DEV_ENABLED=${LOG_FORMAT_LOCAL_DEV_ENABLED}
RABBIT_HOST=${RABBIT_HOST}
RABBIT_PASSWORD=${RABBIT_PASSWORD}
RABBIT_PORT=${RABBIT_PORT}
RABBIT_SECURE=${RABBIT_SECURE}
RABBIT_USER=${RABBIT_USER}
REDIS_HOST=${REDIS_HOST}
REDIS_PORT=${REDIS_PORT}
DIRECTOR_V2_HOST=${DIRECTOR_V2_HOST}
```

### CODE REVIEW

**Review by sanderegg — APPROVED** [review]
Welcome to the dynamic scheduler!

One thing i started doing with new services is to get a starting coverage of >99%. this is very helpful to keep caring for this number being as high as possible.

**Review by matusdrobuliak66 — APPROVED** [review]

File: services/dynamic-scheduler/docker/healthcheck.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+# Adds a base-path if defined in environ
+SIMCORE_NODE_BASEPATH = os.environ.get("SIMCORE_NODE_BASEPATH", "")
+
+
+def is_service_healthy() -> bool:
+    if "debug" in SC_BOOT_MODE.lower():
+        return True
+
+    with suppress(Exception), urlopen(f"{sys.argv[1]}{SIMCORE_NODE_BASEPATH}") as f:
```
Comment (pcrespov) [inline services/dynamic-scheduler/docker/healthcheck.py:35]: We do not need anymore this `SIMCORE_NODE_BASEPATH`. I am not sure we are using this anywhere anymore

File: services/dynamic-scheduler/src/simcore_service_dynamic_scheduler/_meta.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+PROJECT_NAME: Final[str] = info.project_name
+VERSION: Final[Version] = info.version
+API_VERSION: Final[str] = info.__version__
+API_VTAG: Final[str] = info.api_prefix_path_tag
+SUMMARY: Final[str] = info.get_summary()
+
+
+# NOTE: https://patorjk.com/software/taag/#p=display&f=Standard&t=dynamic-scheduler
+APP_STARTED_BANNER_MSG = r"""
+      _                             _                    _              _       _
```
Comment (pcrespov) [inline services/dynamic-scheduler/src/simcore_service_dynamic_scheduler/_meta.py:22]: As @sanderegg ... no real taste for banners ;-)

File: services/dynamic-scheduler/src/simcore_service_dynamic_scheduler/models/schemas/meta.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+from typing import Any, ClassVar
+
+from models_library.basic_types import VersionStr
+from pydantic import BaseModel, HttpUrl
+
+
+class Meta(BaseModel):
```
Comment (pcrespov) [inline services/dynamic-scheduler/src/simcore_service_dynamic_scheduler/models/schemas/meta.py:7]: THOUGHT. This is a model that at some point we might want to put in common so we do not copy it again and again.

File: services/dynamic-scheduler/src/simcore_service_dynamic_scheduler/utils/http_client.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
+import contextlib
+import logging
+
+import httpx
+from fastapi import FastAPI
+
+_logger = logging.getLogger(__name__)
+
+
+class BaseHttpApi:
```
Comment (pcrespov) [inline services/dynamic-scheduler/src/simcore_service_dynamic_scheduler/utils/http_client.py:10]: THOUGHT: this module I will also put in in `serviceslib.fastapi` in a later PR

**Review by pcrespov — APPROVED** [review]
thx!

**Pull request merged**